configs/components/facter.rb

```diff
@@ -155,19 +155,20 @@
     cmake = "/opt/pl-build-tools/i386-pc-solaris2.#{platform.os_version}/bin/cmake"
 
     # FACT-1156: If we build with -O3, solaris segfaults due to something in std::vector
-    special_flags = "-DCMAKE_CXX_FLAGS_RELEASE='-O2 -DNDEBUG'"
+    special_flags += " -DCMAKE_CXX_FLAGS_RELEASE='-O2 -DNDEBUG' "
```

Comment:
So this is where alot of this logic gets SUPER hairy. The osx conditional is actually not concatenated on purpose, as it is supposed to overload the variable, not append to it. What I would like to do is come up with some convention for this concatenation as part of the larger puppet-agent cleanup effort as opposed to doing it here. This PR is already large enough.